### PR TITLE
Add changes from rector rule ReturnEarlyIfVariableRector

### PR DESCRIPTION
**Pull request by Harfusha**
https://github.com/cakephp/cakephp/issues/17756

### CODE REVIEW

**Review by ADmad — APPROVED** [review]

File: src/ORM/ResultSetFactory.php [hunk 1]
```diff
@@ -234,7 +234,7 @@ protected function groupResult(array $row, array $data): EntityInterface|array
             $results = $results[$data['primaryAlias']];
         }
         if ($data['hydrate'] && !($results instanceof EntityInterface)) {
-            $results = new $data['entityClass']($results, $options);
+            return new $data['entityClass']($results, $options);
```
Comment (othercorey) [inline src/ORM/ResultSetFactory.php:239]: This needs an annotation, or we can just revert this change so the normal return is correct.
